### ISSUE TITLE
comments-identation: false positive with commented dictionary in list

### ISSUE THREAD

**Issue by ssbarnea** [adrienverge/yamllint#384]
Here is an example where yamllint reports an error, even if based on the documented examples it should work fine:

```yaml
---
platforms:
  # - name: centos7
  - name: centos
    version: 8
  # - name: debian
  #   version: 10
```

The first commented section is correctly considered well idented but the last one is not. I do mention that this identation is the default one that is applied by IDEs like vscode and assures full reversal via shortcuts (Cmd+/).

As we can see the editor did insert the comment at correct indentation.

I observed that if I removed the line with `version: 8`, yamllint no longer complains about wrong indentation, this should also explain where the bug comes from (assumption that commented line must be part of the previous identation level). 

This is why you have no problems commenting an entire sequence item at start but not if previous line was a mapping.

**Comment by adrienverge** [comment]
Thanks for the report @ssbarnea. Any help to find the cause and fix it (with non-regression tests) would be welcome.

**Comment by ssbarnea** [comment]
I was aware of this bug for longer period of time but I had the impression that you wanted that behavior so I did not bother to investigate a fix. For the moment I changed the defaults of ansible-lint to disable this specific rule.

I am glad that you confirm it as a false-positive, we can now look into addressing it (if we can find time).

**Comment by adrienverge** [comment]
As you point out, this behavior could be wanted in some cases, for example to detect the wrongly-indented comment here:

```yaml
# List of platforms:
platforms:
  - name: centos
    version: 8
  # List of structures:
structures:
  - struct1
```

But since the yamllint error disappears when removing the line `version: 8`, I agree this doesn't really make sense to keep this behavior.

**Comment by vl-kp** [comment]
any workaournd
